src/resources/2015/Dinseylandfill.ts

```diff
@@ -0,0 +1,316 @@
+    $item`none`,
+    $location`Pirates of the Garbage Barges`
+  ),
+  new questData(
+    "racism",
+    3,
+    "Racism Reduction",
+    "Social Justice Adventurer II",
+    "questEStSocialJusticeII",
+    "dinseySocialJusticeIIProgress",
+    15,
+    true,
+    $item`none`,
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`
+  ),
+  new questData(
+    "fun",
+    4,
+    "Compulsory Fun",
+    "Whistling Zippity-Doo-Dah",
+    "questEStZippityDooDah",
+    "dinseyFunProgress",
+    15,
+    true,
+    $item`Dinsey mascot mask`,
+    $location`The Toxic Teacups`
+  ),
+  new questData(
+    "trash",
+    6,
+    "Waterway Debris Removal",
+    "Teach a Man to Fish Trash",
+    "questEStFishTrash",
+    "dinseyFilthLevel",
+    0,
+    true,
+    $item`trash net`,
+    $location`Pirates of the Garbage Barges`
+  ),
+  new questData(
+    "bear",
+    5,
+    "Bear Removal",
+    "Nasty, Nasty Bears",
+    "questEStNastyBears",
+    "dinseyNastyBearsDefeated",
+    8,
+    false,
+    $item`none`,
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`
+  ),
+  new questData(
+    "food",
+    7,
+    "Guest Sustenance Assurance",
+    "Will Work With Food",
+    "questEStWorkWithFood",
+    "dinseyTouristsFed",
+    30,
+    false,
+    $item`none`,
+    $location`Barf Mountain`
+  ),
+];
+
+export function disposeGarbage(): void {
+  if (!get("_dinseyGarbageDisposed") && haveItem($item`bag of park garbage`)) {
+    visitUrl(maintUrl);
+    runChoice(6);
+  }
+}
+
+export function hasQuest(): boolean {
+  for (const quest of quests) {
+    if (quest.currentQuest()) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+export function activeQuest(): questData {
+  for (const quest of quests) {
+    if (quest.currentQuest()) {
+      return quest;
+    }
+  }
+
+  return new questData(
+    "",
+    -1,
+    "",
+    "",
+    "",
+    "",
+    -1,
+    false,
+    $item`none`,
+    $location`none`
+  );
+}
+
+export function questComplete(): boolean {
+  const quest = activeQuest();
+  return hasQuest() && get(quest.questStateProperty) === "finished";
+}
+
+export function hasActiveQuest(): boolean {
+  return hasQuest() && !questComplete();
+}
+
+export function acceptQuest(priority: number | string): void {
+  const page: string = visitUrl(kioskUrl);
+  let choice = 6;
+  const at: number = indexOf(page, "Available Assignments");
+  if (at == -1) {
+    return;
+  }
+
+  const jobs = [
+    `Electrical Maintenance`,
+    `Track Maintenance`,
+    `Sexism Reduction`,
+    `Racism Reduction`,
+    `Compulsory Fun`,
+    `Bear Removal`,
+    `Waterway Debris Removal`,
+    `Guest Sustenance Assurance`,
+  ];
+  let prioritynum = 7;
+  if (typeof priority === "string") {
+    for (const quest of quests) {
+      if (quest.name === priority) {
+        prioritynum = quest.priority;
+        break;
+      }
+    }
+  } else {
+    prioritynum = priority;
+  }
+
+  for (const job1 of jobs.slice(0, prioritynum)) {
```

Comment:
This can also be rewritten more concisely. Also you want to find job1 first and then find job2, they don't need to be in the same loop.

Reply:
I think I've done a nicer job of it now?